linker-interface/shared/src/main/scala/org/scalajs/linker/interface/StandardConfig.scala

```diff
@@ -239,6 +269,100 @@ object StandardConfig {
    */
   def apply(): StandardConfig = new StandardConfig()
 
+  /** Tests whether a string is a valid JS header.
+   *
+   *  A header is valid if and only if it satisfies the following constraints:
+   *
+   *  - It must contain only valid JS whitespace and/or JS comments (single- or multi-line).
+   *  - It must not use new line characters that are not UNIX new lines (`"\n"`).
+   *  - If non-empty, it must end with a new line.
+   *  - It must not contain unpaired surrogate characters (i.e., it must be a valid UTF-16 string).
+   */
+  def isValidJSHeader(jsHeader: String): Boolean = {
+    // scalastyle:off return
+
+    /* First, reject any non-UNIX Unicode new line code point, wherever they
+     * appear (in comments or not). This includes VT and FF, which JavaScript
+     * considers as whitespace but not line separators, as well as NEL.
+     * https://www.unicode.org/reports/tr14/tr14-32.html#BK
+     * VT | FF | CR | NEL | LS | PS
+     *
+     * Also reject unpaired surrogate chars, wherever they appear.
+     */
+    val len = jsHeader.length()
+    var i = 0
+    while (i != len) {
+      def isNewLine(c: Char): Boolean = (c: @switch) match {
+        case '\u000B' | '\u000C' | '\r' | '\u0085' | '\u2028' | '\u2029' => true
+        case _                                                           => false
+      }
+
+      val cp = jsHeader.codePointAt(i)
+      i += Character.charCount(cp)
+      if (Character.isBmpCodePoint(cp)) {
+        if (isNewLine(cp.toChar) || Character.isSurrogate(cp.toChar))
+          return false
+      }
+    }
+
+    // Now, parse whitespace and comments, and reject anything else
+
+    i = 0
+    while (i != len) {
+      val cp = jsHeader.codePointAt(i)
+      i += Character.charCount(cp)
+
+      (cp: @switch) match {
+        // Accept the UNIX new line
+        case '\n' =>
+          ()
```

Comment:
Using a unit to do nothing in statement position is quite odd IMHO. Do we have precedent for that in SJS? Just put nothing?

Reply:
I have started using that instead of an `// OK` comment (which is what we used in the past) for future-proofing against scalafmt. I have noticed in other repos that scalafmt aligns a comment on its own line at the end of the `case` with the *following* `case` statement (hence 2 spaces to the left). The `()` prevents that, and also stands on its own as "yes, I intended to leave this statement empty".